ingestclient/core/engine.py

```diff
@@ -341,6 +341,8 @@ def monitor(self, workers):
                         log_str = "Uploading ~{:.2f} {}/min".format(
                             avg_tile_rate * 6, units
                         )
+                        if avg_tile_rate == 0:
+                            avg_tile_rate = 1  # don't want divide by zero
                         remaining_min = status["current_message_count"] / avg_tile_rate
```

Comment:
I would replace with: 
```python
                        remaining_min = None if avg_tile_rate == 0 else status["current_message_count"] / avg_tile_rate
                        if remaining_min is None:
                            eta_text = "calculating ETA..."
                        elif remaining_min > 60*24:
                            eta_text = "ETA: {:.2f} days".format(remaining_min / (60 * 24))
                        elif remaining_min > 60:
                            eta_text = "ETA: {:.2f} hours".format(remaining_min / 60)
                        else:
                            eta_text = "ETA: {:.2f} minutes".format(remaining_min)
```

because no progress is being made if the tile rate is 0.

Reply:
I like it!.  I incorporated your suggestion.